projects/make-recall-decisions-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/casesummary/Contact.kt

```diff
@@ -0,0 +1,167 @@
+import java.time.LocalDate
+import java.time.ZonedDateTime
+
+@Immutable
+@Table(name = "contact")
+@Entity(name = "CaseSummaryContact")
+@Where(clause = "soft_deleted = 0")
+class Contact(
+    @Id
+    @Column(name = "contact_id")
+    val id: Long,
+
+    @Column(name = "offender_id")
+    val personId: Long,
+
+    @Column
+    val description: String?,
+
+    @ManyToOne
+    @JoinColumn(name = "contact_type_id")
+    val type: ContactType,
+
+    @ManyToOne
+    @JoinColumn(name = "contact_outcome_type_id")
+    val outcome: ContactOutcome? = null,
+
+    @OneToMany(mappedBy = "contact")
+    val documents: List<ContactDocument>,
+
+    @Lob
+    @Column
+    val notes: String? = null,
+
+    @Column(name = "contact_date")
+    val date: LocalDate,
+
+    @Column(name = "contact_start_time")
+    val startTime: ZonedDateTime,
+
+    @Column
+    @Convert(converter = YesNoConverter::class)
+    val sensitive: Boolean,
+
+    @Column(name = "soft_deleted", columnDefinition = "number")
+    val softDeleted: Boolean = false
+)
+
+@Immutable
+@Table(name = "r_contact_type")
+@Entity(name = "CaseSummaryContactType")
+class ContactType(
+    @Id
+    @Column(name = "contact_type_id")
+    val id: Long,
+
+    @Column
+    val code: String,
+
+    @Column
+    val description: String,
+
+    @Column(name = "sgc_flag", columnDefinition = "number")
+    val systemGenerated: Boolean
+)
+
+@Immutable
+@Table(name = "r_contact_outcome_type")
+@Entity(name = "CaseSummaryContactOutcome")
+class ContactOutcome(
+    @Id
+    @Column(name = "contact_outcome_type_id")
+    val id: Long,
+
+    @Column
+    val description: String
+)
+
+@Immutable
+@Table(name = "document")
+@Entity(name = "CaseSummaryContactDocument")
+@Where(clause = "table_name = 'CONTACT' and soft_deleted = 0")
+class ContactDocument(
+    @Id
+    @Column(name = "document_id")
+    val id: Long,
+
+    @ManyToOne
+    @JoinColumn(name = "primary_key_id", referencedColumnName = "contact_id")
+    val contact: Contact,
+
+    @Column(name = "document_name")
+    val name: String,
+
+    @Column
+    val tableName: String = "CONTACT",
+
+    @Column(name = "soft_deleted", columnDefinition = "number")
+    val softDeleted: Boolean = false
+)
+
+interface CaseSummaryContactRepository : JpaRepository<Contact, Long> {
+    @Query(
+        """
+        select c from CaseSummaryContact c
+        left join fetch c.type
+        left join fetch c.outcome
+        left join fetch c.documents
+        where c.personId = :personId
+        and c.date <= :toDate
+        and (:fromDate is null or c.date >= :fromDate)
+        and (:typesCount = 0 or c.type.code in :types)
+        and (:includeSystemGenerated = 1 or c.type.systemGenerated = false)
+        order by c.date desc, c.startTime desc
+        """
+    )
+    fun findContacts(
+        personId: Long,
+        fromDate: LocalDate?,
+        toDate: LocalDate,
+        includeSystemGenerated: Int,
```

Comment:
It actually seemed to be a Hibernate/H2 issue - it works when comparing column values to true/false (I guess because it has the column definition / converter defined), but when comparing input parameters to true/false it failed.  I tried a few things, but explicitly converting to an Int was the only thing that worked in the end.

Well this is annoying... I just reverted it to a Boolean locally so I could get you the error message but this time it works 🤯 I'll double check, but if it's magically started working I'll revert it to a Boolean.